extension/lib/logger.js

```diff
@@ -108,24 +103,34 @@ Logger.prototype._writeAsync = function Logger_writeAsync(aMessage, aCallback) {
   // For testing purposes send the message to stdout
   //dump(aMessage + '\n');
 
-  // Create an output stream to write to file
-  var foStream = Cc["@mozilla.org/network/file-output-stream;1"]
-                 .createInstance(Ci.nsIFileOutputStream);
-  foStream.init(this._file, 0x02 | 0x08 | 0x10, PERMS_FILE, foStream.DEFER_OPEN);
-
-  // Write asynchronously to buffer;
-  // Input and output streams are closed after write
-  var iStream = this._converter.convertToInputStream(aMessage);
-  NetUtil.asyncCopy(iStream, foStream, function (status) {
-    if (!components.isSuccessCode(status)) {
-      var errorMessage = new Error("Error while writing to file: " + status);
-      console.error(errorMessage);
+  function callback(aResult) {
+    if (typeof(aCallback) === "function") {
+      aCallback(aResult);
```

Comment:
I tried it without success. OK thats my bad, I'm promise noob. Maybe next time. :)